### PR TITLE
Implement `!global` assignments

### PR DESCRIPTION
**Pull request by mgreter**
Implemented correct `!global` and `!default` handling. Was actually not that hard once I "understood" how it all ought to work, but `@for` and `@each` threw me off a bit, since they don't seem to be "real" blocks (they don't seem to have their own env scope). Added a few additional spec tests which should cover some of the more advanced use cases (which haven't been covered yet). https://github.com/sass/sass-spec/pull/282

Oh and btw. this fixes https://github.com/sass/libsass/issues/613

### CODE REVIEW

**Comment by mgreter** [comment]
@xzyfer maybe you want to review? IMO it's ready to be merged!

**Comment by xzyfer** [comment]
Sorry I've been offline running a conference. I'll catch up today.

File: functions.cpp [hunk 1]
```diff
@@ -145,7 +145,9 @@ namespace Sass {
     static mt19937 rand(static_cast<unsigned int>(GetSeed()));
 
     // features
-    static set<string> features;
+    static set<string> features {
+      "global-variable-shadowing"
```
Comment (xzyfer) [inline functions.cpp]: We should remember to mention this in the changelog.

File: debugger.hpp [hunk 2]
```diff
@@ -69,7 +69,7 @@ inline void debug_ast(AST_Node* node, string ind = "", Env* env = 0)
         case Complex_Selector::ADJACENT_TO: cerr << "{+}"; break;
```
Comment (xzyfer) [inline debugger.hpp]: Cool. I'd fixed this in a branch that I haven't PR'd yet :)

**Comment by xzyfer** [comment]
LGTM, nice one

File: debugger.hpp (hunk 2 above)
Comment (mgreter) [inline debugger.hpp]: Looks like your using `debug_ast` too :wink: I hope you like it as much as I do :smile: It actually decreased my time hunting bugs dramatically! I also thought about creating a nicer `debug_env` function, but I was able to use existing one (altough the output could be better there).

Comment (xzyfer) [inline debugger.hpp]: I saw you post debub_ast output in an issue and fell in love.

**Pull request merged**